### DISCUSSION TITLE
MySQL variables

### DISCUSSION THREAD

**Discussion by ministerofpanic** [planetscale/discussion#287] [Q&A] [answered]
Hello 👋🏻 

I'm trying to use a variable to track which row I'm on in my table but I'm getting a syntax error (near ':'), can someone help point out my mistake please?

```
SET @counter = 0;

SELECT
    @counter
FROM
    foo,
    (SELECT @counter:=@counter+1);
```

**Expected behaviour:**
`@counter` will start at 0 and increment by 1 for each row in `foo`

**Comment by orware** [comment, marked as answer]
Hi @artdevgame,

I'm not sure if the syntax error is due to one of our listed [MySQL Compatibility](https://planetscale.com/docs/reference/mysql-compatibility) entries since it may be that the [assignment operator](https://dev.mysql.com/doc/refman/8.0/en/assignment-operators.html) (e.g. `:=` ) may not be allowed at this time.

An alternative that comes to mind however would be to make use of the `ROW_NUMBER()` [Window Function](https://dev.mysql.com/doc/refman/8.0/en/window-function-descriptions.html#function_row-number).

```
SELECT ROW_NUMBER() OVER (ORDER BY id) AS 'row_number'
FROM foo;
```
(Assuming your `foo` table has a column named `id` to use within the `OVER` [clause](https://dev.mysql.com/doc/refman/8.0/en/window-functions-usage.html))

**Reply by ministerofpanic** [reply]
Awesome, that worked @orware, thanks!

**Reply by orware** [reply]
You're welcome @artdevgame I'm happy that solution worked for your needs!

Also, I was able to find the relevant Vitess issue that talks about the assignment operator usage limitation:
https://github.com/vitessio/vitess/issues/6595
